gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/InstrumentatedGoogleHadoopFileSystem.java

```diff
@@ -308,4 +316,39 @@ public IOStatistics getIOStatistics() {
   public GhfsInstrumentation getInstrumentation() {
     return this.instrumentation;
   }
+
+  /** Set the Value for http get and head request related statistics keys */
+  private void setHttpStatistics() {
+    try {
+      if (getGcsFs() != null) {
+        try {
+          long get_failures =
+              getGcsFs()
+                  .getGcs()
+                  .getStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+                  .longValue();
+          if (get_failures > 0L) {
+            instrumentation.incrementFailureStatistics(
+                GhfsStatistic.ACTION_HTTP_GET_REQUEST.getSymbol(), get_failures);
+          }
+        } catch (Exception e) {
+          logger.atWarning().log("GetStatistics from gcsio threw Unsupported Exception");
+        }
+      }
+
+      if (getGcsFs().getGcs().getGcsRequestsTracker() != null) {
+        ImmutableList<HttpRequest> httpRequests =
+            this.getGcsFs().getGcs().getGcsRequestsTracker().getAllRequests();
+        for (HttpRequest req : httpRequests) {
+          if (req.getRequestMethod() == "GET") {
+            instrumentation.incrementCounter(GhfsStatistic.ACTION_HTTP_GET_REQUEST, 1);
+          } else if (req.getRequestMethod() == "HEAD") {
+            instrumentation.incrementCounter(GhfsStatistic.ACTION_HTTP_HEAD_REQUEST, 1);
+          }
+        }
```

Comment:
How about we group by method and increment the count only once ?

Reply:
Updated